### PR TITLE
refactor(test): convert tests to mocha

### LINKED ISSUE #260: Tracking issue: Convert ava to mocha in tests
- [ ] test-clean.ts
- [ ] test-format.ts
- [ ] test-init.ts
- [ ] test-kitchen.ts
- [ ] test-lint.ts
- [ ] test-util.ts

### PR DESCRIPTION
**Pull request by JustinBeckwith**
Fixes #260.  @ofrobots I did most of the grunt work here, and now I'm just fighting a few failing tests.  I pushed this to upstream so we could maybe collaborate on this one.   

Side note: really suggest enabling 'hide whitespace changes' for this one

### CODE REVIEW

**Comment by ofrobots** [comment]
Awesome! I will take a look when I get a chance.

**Comment by vitkarpov** [comment]
Out of curiosity, why Mocha and not Jest? 😄

**Comment by JustinBeckwith** [comment]
@vitkarpov honestly, just consistency.  We use `mocha` for everything else, so this just makes it easier to not think and get stuff done.

**Comment by vitkarpov** [comment]
Gotcha, it's totally fine, I'm just asking 😉

**Comment by vitkarpov** [comment]
@JustinBeckwith which codemode are you using to apply this transition?

**Comment by JustinBeckwith** [comment]
There were some really nasty merge conflicts on this one, so please bear with me.  🐻

**Comment by JustinBeckwith** [comment]
@google/google-node-team this is ready for a look!

File: test/test-format.ts [hunk 1]
```diff
@@ -132,116 +139,119 @@ test.serial(
...
+  it('format globs listed in include', () => {
+    return withFixtures(
+      {
+        'tsconfig.json': JSON.stringify({ include: ['dirb/*'] }),
+        dira: { 'a.ts': GOOD_CODE },
+        dirb: { 'b.ts': BAD_CODE },
+      },
+      async () => {
+        const okay = await format.format(OPTIONS);
+        assert.strictEqual(okay, false);
```
Comment (kjin) [inline test/test-format.ts]: Nit -- I see both `assert.strictEqual(okay, false)` and `assert.ok(!okay)`, can we just use one of those?

File: test/test-init.ts [hunk 2]
```diff
@@ -58,124 +56,126 @@ function hasExpectedDependencies(packageJson: PackageJson): boolean {
...
-    },
-    async () => {
-      const result = await init.init(OPTIONS_YARN);
-      t.truthy(result);
-
-      const contents = await readJson('./package.json');
-      t.truthy(contents.scripts.prepare === 'yarn run compile');
-    }
-  );
+  it('init should support yarn', async () => {
```
Comment (kjin) [inline test/test-init.ts]: Nit: this method has `async () => { return Promise }`, I've also seen `async () => { await Promise }` and `() => { return Promise }`, they're all valid but should we stick with just one pattern?

File: test/test-init.ts [hunk 3]
```diff
@@ -58,124 +56,126 @@ function hasExpectedDependencies(packageJson: PackageJson): boolean {
...
+      {
+        'package.json': JSON.stringify({ name: 'test' }),
+        'yarn.lock': '',
+      },
+      async () => {
+        const result = await init.init(OPTIONS_YARN);
+        assert.ok(result);
+
+        const contents = await readJson('./package.json');
+        assert.ok(contents.scripts.prepare === 'yarn run compile');
```
Comment (kjin) [inline test/test-init.ts]: Can we use `strictEqual` here?

File: test/test-kitchen.ts [hunk 4]
```diff
@@ -66,140 +62,138 @@ const execOpts = {
...
- * CLEAN UP - remove the staging directory when done.
- */
-test.after.always('cleanup staging', async () => {
-  if (!keep) {
-    stagingDir.removeCallback();
-  }
+  /**
+   * CLEAN UP - remove the staging directory when done.
+   */
+  after('cleanup staging', () => {
```
Comment (kjin) [inline test/test-kitchen.ts]: Nit: Should `after` be right after `before`? I think that's our "standard" in other tests. Could go either way though

File: test/test-lint.ts [hunk 5]
```diff
@@ -25,103 +25,108 @@ import { nop } from '../src/util';
...
-});
+  it('lint should return true on good code', () => {
+    return withFixtures(
+      {
+        'tsconfig.json': JSON.stringify({ files: ['a.ts'] }),
+        'a.ts': GOOD_CODE,
+      },
+      async () => {
+        const okay = lint.lint(OPTIONS);
+        assert.strictEqual(okay, true);
```
Comment (kjin) [inline test/test-lint.ts]: Here and elsewhere, use `assert.ok`?

File: test/test-format.ts (hunk 1 above)
Comment (JustinBeckwith) [reply test/test-format.ts]: yeah that's gross, good feedback

File: test/test-lint.ts (hunk 5 above)
Comment (JustinBeckwith) [reply test/test-lint.ts]: I went the other way.  I prefer to use `assert.strictEqual` to verify the object returned is actually a `bool`, and not a random object or something.  I made this consistent throughout.

**Comment by JustinBeckwith** [comment]
@kjin this is ready for another look!

File: test/test-lint.ts (hunk 5 above)
Comment (kjin) [reply test/test-lint.ts]: We already get guarantees on return types from TypeScript, IMO we should not duplicate that check in our tests. But this is fine, the outcome is the same.

**Pull request merged**